OSDOCS-5074: Choosing subnets when creating a LoadBalancerService ingress Controller (#85350)
openshift/openshift-docs: modules/nw-ingress-setting-update-subnet-LoadBalancerService.adoc

Reviewer: This filename and ID also need to be lowercase

// Modules included in the following assemblies:
//
// * ingress/configure-ingress-operator.adoc

:_mod-docs-content-type: PROCEDURE
[id="nw-ingress-setting-update-subnet-LoadBalancerService_{context}"]

> = Updating the subnets on an existing Ingress Controller
> You can update an `IngressController` with manually specified load balancer subnets in {product-title} to avoid any disruptions, to maintain the stability of your services, and to ensure that your network configuration aligns with your specific requirements. The following procedures show you how to select and apply new subnets, verify the configuration changes, and confirm successful load balancer provisioning. 

Reviewer: ```suggestion
= Updating the subnets on an existing Ingress controller

You can update an Ingress controller with manually specified load balancer subnets in {product-title} to avoid any disruptions, to maintain the stability of your services, and to ensure that your network configuration aligns with your specific requirements. The following procedures show you how to select and apply new subnets, verify the configuration changes, and confirm successful load balancer provisioning. 
```

That first sentence is doing a lot :) can it be broken up?

We tend to want to avoid self-referential language, would it make sense to just remove the second sentence? Or rewrite as things the user can do? (i.e. "You can use the {oc-first} to select and apply new subnets, verify the configuration changes, and confirm successful load balancer provisioning." or similar)

> .Procedure
> To update an `IngressController` with manually specified load balancer subnets, you can follow these steps:
> 
> . Choose the subnets.

Reviewer: ```suggestion
.Procedure

. Choose the subnets.
```

> . Choose the subnets.

Reviewer: What are they actually doing here? Editing a YAML file? How do they find it?

+
[source,yaml]
----
apiVersion: operator.openshift.io/v1
kind: IngressController
metadata:
  name:  <name> <1>
  namespace: openshift-ingress-operator
spec:
  domain: <domain> <2>
  endpointPublishingStrategy:
    type: LoadBalancerService
    loadBalancer:
      scope: External
      providerParameters:
        type: AWS
        aws:
          type: Classic <3>
          classicLoadBalancer: <4>
            subnets:
              ids: <5>
              - <updated_subnet> <6>
              - <updated_subnet>
              - <updated_subnet>
----
<1> Replace `<name>` with a name for the `IngressController`.
<2> Replace `<domain>` with the DNS name serviced by the `IngressController`.

> <3> Specify updated subnet IDs (or names if you using `names`).

Reviewer: Same feedback here as where they appear in the other module (avoid using things in backticks as nouns etc)

<4> You can also use the `networkLoadBalancer` field if using an NLB.
<5> You can optionally specify a subnet by name using the `names` field instead of specifying the subnet by ID.

> <6> Specify subnet IDs (or names if you using `names`).

Reviewer: ```suggestion
<6> Specify subnet IDs (or names if using the `names` field).
```

> . Examine the `Progressing` condition on the `IngressController` for instructions on how to apply the subnet updates:

Reviewer: ```suggestion
. Examine the `Progressing` condition on the `IngressController` for instructions on how to apply the subnet updates by running the following command:
```

+
[source,terminal]
----
$ oc get ingresscontroller -n openshift-ingress-operator subnets -o jsonpath="{.status.conditions[?(@.type==\"Progressing\")]}" | yq -PC
----
+
.Example output
[source,terminal]
----
lastTransitionTime: "2024-11-25T20:19:31Z"
message: 'One or more status conditions indicate progressing: LoadBalancerProgressing=True (OperandsProgressing: One or more managed resources are progressing: The IngressController subnets were changed from [...] to [...].  To effectuate this change, you must delete the service: `oc -n openshift-ingress delete svc/router-<name>`; the service load-balancer will then be deprovisioned and a new one created. This will most likely cause the new load-balancer to have a different host name and IP address and cause disruption. To return to the previous state, you can revert the change to the IngressController: [...]'
reason: IngressControllerProgressing
status: "True"
type: Progressing
----

> . Delete the service associated with the `IngressController` to apply the update:

Reviewer: ```suggestion
. To apply the update, delete the service associated with the Ingress controller by running the following command:
```

+
[source,terminal]
----
$ oc -n openshift-ingress delete svc/router-<name>
----
+
[WARNING]
====

> An outage that can last several minutes due to new DNS records propagation, new load balancers provisioning, and other factors. IP addresses and canonical names of the Ingress Controller load balancer might change after applying this procedure.

Reviewer: First sentence seems to be missing a noun or something, not sure what is needed to fix it.
```suggestion
An outage that can last several minutes due to new DNS records propagation, new load balancers provisioning, and other factors. IP addresses and canonical names of the Ingress controller load balancer might change after applying this procedure.
```

Author: Agreed. I think the problem is the "that" that shouldn't be there.

====

> . Confirm the load balancer was provisioned successfully by checking the `IngressController` conditions.
> +
> [source,terminal]
> ----  
> $ oc get ingresscontroller -n openshift-ingress-operator <name> -o jsonpath="{.status.conditions}" | yq -PC
> ----

Reviewer: ```suggestion
.Verification

* To confirm that the load balancer was provisioned successfully, check the `IngressController` conditions by running the following command:
+
[source,terminal]
----  
$ oc get ingresscontroller -n openshift-ingress-operator <name> -o jsonpath="{.status.conditions}" | yq -PC
----
```
